### ISSUE TITLE
Nueva funcion encriptación con N

### ISSUE THREAD

**Issue by terecarvallo** [IIC2233/Syllabus#466]
### Prerrequisitos
(Marcar colocando una X entre los corchetes los ítems que ya hiciste, así: "[X]")

* [ X] Leí las reglas del foro (https://github.com/IIC2233/Syllabus/wiki/8.-Reglas-del-foro-(issues)).
* [X] Busqué en las *issues* si ya preguntaron mi duda y no encontré nada parecido (https://github.com/IIC2233/syllabus/issues/).
* [X] Revisé el compilado de dudas de la tarea y no encontré una *issue* similar a la mía ([click aquí para ir al compilado](/IIC2233/syllabus/issues/291)).
* [X] Mi duda no se trata sobre una librería, *built-in* o mala práctica, ya que eso se pregunta en la *issue* creada para ello ([click aquí para ir a esa _issue_](/IIC2233/syllabus/issues/290)).
* [X] Mi duda no se trata de un tema administrativo o personal, ya que en ese caso debo contactar a mi profe, al ayudante de Bienestar o al correo del curso (iic2233@ing.puc.cl).
* [X] Utilizaré un título descriptivo y llenaré correctamente esta plantilla.
* [X] De ser necesario, colocaré código simple que permita explicar mi problema o duda, ya que compartir código de esta evaluación puede ser considerado una falta a la ética.

En el nuevo enunciado, hay que definir un nuevo parámetro, el cual es el único que se usa para encriptar el mensaje. Pero en cripto.py encriptar y desencriptar tienen un argumento N que es un int, para que se usa este? En el enunciado no se menciona.

**Comment by MelanyLillo** [comment]
Hola! Ese N se refiere a N_PONDERADOR, el cual se estableció como N=1 en cripto .py para testear.
Avísame si se resolvió tu duda :)

**Comment by tremendapithon** [comment]
Hola buenas, tengo una duda de que pasa si el ponderador N es mayor al largo del mensaje encriptado, que deberia hacer en esos casos?

**Comment by tremendapithon** [comment]
Pense en hacer una función que verifique en que posición deberia quedar si N > len(msg_encriptado)

**Comment by MelanyLillo** [comment]
Hola!
Si N es mayor al largo del mensaje simplemente se siguen moviendo los bytes dando las vueltas que sean necesarias. De hecho podrías hacer una división entre N y el largo del mensaje para notar las vueltas completas que habrán y así evitar hacer esos movimientos innecesarios.

Ejemplo:

tengo el bytearray ABC, y mi N_ponderador da 7. Entonces

Movimiento 1: CAB
Movimiento 2: BCA
Movimiento 3: ABC
Movimiento 4: CAB
Movimiento 5: BCA
Movimiento 6: ABC
Movimiento 7: CAB

Entonces el resultado final es CAB

Avísame si se respondió tu duda😊

**Comment by tremendapithon** [comment]
Osea me referia al momento de intercambiar posiciones, eso me falto

**Comment by tremendapithon** [comment]
Tipo después de terminar los movimientos hay que intercambiar el primer carácter con el carácter N, pero si N > len(msg), qué pasa?
Tengo una idea, pero quería saber que recomiendan

**Comment by MelanyLillo** [comment]
En el enunciado se explica que deben usar módulo, de este modo, si N es 10 y el largo es 4  entonces 10 mod 4 = 2.

Ejemplo

Si tengo el mensaje ABCD y quiero el N = 6
A B C D
0 1 2 3
4 5 6

Entonces la letra en N=6 es la C
Avísame si se respondió tu pregunta ☺️

**Comment by tremendapithon** [comment]
modulo?

**Comment by tremendapithon** [comment]
este ```%```?

**Comment by tremendapithon** [comment]
creo que ya tengo una idea, gracias

**Comment by tremendapithon** [comment]
@MelanyLillo entonces lo de 10 mod 4 = 2, el 2 significa la cantidad de veces que nos vamos a mover?

**Comment by MelanyLillo** [comment]
No, eso es para el tercer paso de la encriptación, donde se solicita intercambiar valores.

**Comment by tremendapithon** [comment]
@MelanyLillo intente hacer eso, pero me dio error de unpickling

**Comment by MelanyLillo** [comment]
Tu error podría estar en que estás entregando un objeto con una cantidad mayor de bytes de lo que es capaz de recibir, revisa el largo del mensaje enviado y entregado y si estos concuerdan, puesto que si el largo de tu mensaje es mayor que el aceptado, se cortará.

**Comment by tremendapithon** [comment]
@MelanyLillo Estaba viendo y me da problemas cuando el ponderador es un multiplo de 10, incluso probe con 164 y no tuve problemas.

**Comment by MelanyLillo** [comment]
Viendo tu código solo se me ocurre que podría ser que no está siguiendo fielmente la fórmula, pues en el enunciado muestra que es bytearray_original[Y] = bytearray_encriptado[(Y + N) mod len(bytearray_original)], pero no estás considerando Y.

**Comment by tremendapithon** [comment]
@MelanyLillo Pero es raro, ya que para casos en donde el ponderador no es un multiplo de 10, si funciona

**Comment by tremendapithon** [comment]
@MelanyLillo Pero no se que es Y, osea yo lo hice considerando el movimiento de los bytes, para eso utilizo un iterador

**Comment by tremendapithon** [comment]
@MelanyLillo Por lo que entiendo de "bytearray_original[Y] = bytearray_encriptado[(Y + N) mod len(bytearray_original)]" es que el bytearray de Y tiene que quedar en la posicion (Y + N) mod len(bytearray) y eso lo termino haciendo bien, ya que el test de encriptacion si me funciona

**Comment by tremendapithon** [comment]
Eso hago, si el mensaje llega bien, pero no encuentro el sentido que no funcione con multiplos de 10.

**Comment by MelanyLillo** [comment]
Puede ser que justo envía algún mensaje de largo 10 (o múltiplo de 10) y eso hace que quizás se cae el código por algo que no está implementado correctamente 🤔 Por ahora te recomiendo dejar N_PONDERADOR con otro valor distinto a un múltiplo de 10, y avanzar con otras cosas de la tarea para evitar perder tiempo en eso. En el peor de los casos puedes dejar un mensaje sobre esto en tu readme para que lo tengamos en consideración al momento de evaluar.

**Comment by MelanyLillo** [comment]
Buenísimo, éxito en tu tarea!

**Comment by martineyz** [comment]
Respecto a éste tema, en el enunciado se dice explicitamente que se intercambia el valor de bytearray_encripatado en la posicion N por el valor en la posicion 0. Enonces si N es mayor o igual al largo del mensaje se daría un index error. Según el enunciado el modulo se usa para determinar la nueva posicion de un byte cualquiera en la posicion Y, pero no habla nada de usar el modulo a la hora de inercambiar la posicion 0 con la posicion N. Me gustaría reibir una aclaración respecto a esto. Gracias.

**Comment by MelanyLillo** [comment]
Hola! La idea es que aquí también usen módulo para evitar el error que mencionas, en una respuesta anterior puse un ejemplo de cómo funcionaría esto.
Avísame si se respondió tu pregunta ☺️
